src/main/java/module-info.java

```diff
@@ -0,0 +1,14 @@
+module de.swiesend.secretservice {
+    requires java.desktop;
+    requires org.freedesktop.dbus;
+    requires org.slf4j;
+    requires at.favre.lib.hkdf;
+
+    opens de.swiesend.secretservice to org.freedesktop.dbus;
+
+    exports de.swiesend.secretservice;
+    exports de.swiesend.secretservice.simple;
+    exports org.freedesktop.secret.interfaces;
+    exports org.freedesktop.secret.simple.interfaces;
+    exports org.gnome.keyring.interfaces;
```

Comment:
In order to rule out split packages, these need to be renamed as well. (Issue #38)

Reply:
I know, this is intended for two reasons:

- the D-Bus interfaces require to be extended by Java interfaces, to be used:
  https://github.com/purejava/secret-service/blob/e3f20f4e144fe2141fb28ed92ec75ef321b618e3/src/main/java/org/freedesktop/secret/interfaces/Collection.java#L19

  https://github.com/purejava/secret-service/blob/e3f20f4e144fe2141fb28ed92ec75ef321b618e3/src/test/resources/org.freedesktop.Secret.Collection.xml#L5

  The package name of the Java interface needs to follow the interface name of the D-Bus interface, as D-Bus responses are recognized and differentiated by their interface name: https://github.com/purejava/secret-service/blob/e3f20f4e144fe2141fb28ed92ec75ef321b618e3/src/main/java/org/freedesktop/secret/handlers/MessageHandler.java#L47
  
  See [D-Bus Java Documtation](https://dbus.freedesktop.org/doc/dbus-java/dbus-java/dbus-javase3.html#x13-160003.1), first paragraph.
  
- this avoids programmers from implementing two libraries handling the same D-Bus interfaces with different implementations on the library side wich would cause an appropriate warning message from the compiler

Reply:
This is a problem. But your linked docs suggest a solution as well:

> It is highly recommended that the Java interface and package name match the D-Bus interface. However, if, for some reason, this is not possible then the name can be overridden by use of an Annotation.

Reply:
> This is a problem. But your linked docs suggest a solution as well:
> 
> > It is highly recommended that the Java interface and package name match the D-Bus interface. However, if, for some reason, this is not possible then the name can be overridden by use of an Annotation.

That sounds reasonable, and I will give it a try.

Reply:
Solved by https://github.com/swiesend/secret-service/pull/40/commits/24f080dcf76acd82a804f1ea6f02e7f112f56d87.

It wasn't necessary to rewrite of the D-Bus message handling. It was only necessary to export all interfaces in the `module-info.java` to avoid these:
```java
Caused by: java.lang.IllegalAccessException: class org.freedesktop.dbus.messages.DBusSignal (in module org.freedesktop.dbus) cannot access class de.swiesend.secretservice.interfaces.Collection$ItemCreated (in module de.swiesend.secretservice) because module de.swiesend.secretservice does not export de.swiesend.secretservice.interfaces to module org.freedesktop.dbus
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:394)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:709)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:493)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:484)
	at org.freedesktop.dbus@4.3.0/org.freedesktop.dbus.messages.DBusSignal.createReal(DBusSignal.java:237)
	... 4 common frames omitted
```
`module-info.java`now includes all necessary exports.